components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -887,6 +899,21 @@ void ESP8266::attach(Callback<void()> status_cb)
     _conn_stat_cb = status_cb;
 }
 
+void ESP8266::attach_oob_busy_cb(mbed::Callback<void()> cb)
+{
+    _oob_busy_register_timeout_cb = cb;
+}
+
+void ESP8266::oob_busy_timeout()
+{
+    tr_debug("oob_busy_timeout called");
+    if (_busy)
```

Comment:
Unnecessary check because busy has occurred if this function gets called

Reply:
Yes, but we might have gotten "busy..", registered the callback and then managed to actually send the data, which would clear the _busy flag. Does this make sense?

Reply:
But the lock is released in between?